nifi-nar-bundles/nifi-oauth-bundle/nifi-oauth/src/main/java/org/apache/nifi/oauth/AbstractOAuthControllerService.java

```diff
@@ -0,0 +1,172 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.nifi.oauth;
+
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+
+import org.apache.nifi.components.PropertyDescriptor;
+import org.apache.nifi.controller.AbstractControllerService;
+import org.apache.nifi.controller.ConfigurationContext;
+import org.apache.nifi.processor.util.StandardValidators;
+import org.apache.nifi.reporting.InitializationException;
+
+
+public abstract class AbstractOAuthControllerService
+    extends AbstractControllerService implements OAuth2ClientService {
+
+    protected String accessToken = null;
+    protected String refreshToken = null;
+    protected String tokenType = null;
+    protected long expiresIn = -1;
+    protected long expiresTime = -1;
+    protected long lastResponseTimestamp = -1;
+    protected Map<String, String> extraHeaders = new HashMap<String, String>();
+    protected String authUrl = null;
+    protected long expireTimeSafetyNetSeconds = -1;
+    protected String accessTokenRespName = null;
+    protected String expireTimeRespName = null;
+    protected String expireInRespName = null;
+    protected String tokenTypeRespName = null;
+    protected String scopeRespName = null;
+
+    public static final PropertyDescriptor AUTH_SERVER_URL = new PropertyDescriptor
+            .Builder().name("OAuth2 Authorization Server URL")
+            .displayName("OAuth2 Authorization Server")
+            .description("OAuth2 Authorization Server that grants access to the protected resources on the behalf of the resource owner.")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RESPONSE_ACCESS_TOKEN_FIELD_NAME = new PropertyDescriptor
+            .Builder().name("JSON response 'access_token' name")
+            .displayName("JSON response 'access_token' name")
+            .description("Name of the field in the JSON response that contains the access token. IETF OAuth2 spec default is 'access_token' if your API provider's" +
+                    " response field is different this is where you can change that.")
+            .defaultValue("access_token")
+            .required(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor RESPONSE_EXPIRE_TIME_FIELD_NAME = new PropertyDescriptor
+            .Builder().name("JSON response 'expire_time' name")
```

Comment:
The `name` should avoid special characters and spaces for serialization purposes. Suggest `.name("JSON_response_expire_time_name")`.